test/extended/openstack/loadbalancers.go

```diff
@@ -258,6 +260,77 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb][Serial] The O
+
+			g.By("Checking cluster configuration")
+			setting, err := getClusterLoadBalancerSetting("max-shared-lb", cloudProviderConfig)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			maxSharedLb, err := strconv.Atoi(setting)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if maxSharedLb < 2 {
+				e2eskipper.Skipf("Test not applicable when max-shared-lb is lower than 2 and it is %d", maxSharedLb)
+			}
+
+			g.By("Creating Openshift deployment")
+			labels := map[string]string{"app": "udp-lb-shared-dep"}
+			testDeployment := createTestDeployment("udp-lb-shared-dep", labels, 2, v1.ProtocolUDP, 8081)
+			deployment, err := clientSet.AppsV1().Deployments(oc.Namespace()).Create(ctx,
+				testDeployment, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Creating first Openshift service")
+			svcName1 := "udp-lb-shared1-svc"
+			svcPort1 := int32(8082)
+			jig := e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName1)
+			jig.Labels = labels
+			svc1, err := jig.CreateLoadBalancerService(ctx, loadBalancerServiceTimeout, func(svc *v1.Service) {
+				svc.Spec.Ports = []v1.ServicePort{{Protocol: v1.ProtocolUDP, Port: svcPort1, TargetPort: intstr.FromInt(8081)}}
+				svc.Spec.Selector = labels
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			loadBalancerId := svc1.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
+			o.Expect(loadBalancerId).ShouldNot(o.BeEmpty(), "load-balancer-id annotation missing")
+			e2e.Logf("detected loadbalancer id is %q", loadBalancerId)
+
+			g.By(fmt.Sprintf("Creating internal Openshift service using the existing Openstack loadbalancer %s -"+
+				"This should not be allowed.", loadBalancerId))
+			svcName2 := "udp-lb-shared2-internal-svc"
+			svcPort2 := int32(8083)
+			jig = e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName2)
+			jig.Labels = labels
+			svc2, err := jig.CreateLoadBalancerService(ctx, 1*time.Second, func(svc *v1.Service) {
+				svc.Spec.Ports = []v1.ServicePort{{Protocol: v1.ProtocolUDP, Port: svcPort2, TargetPort: intstr.FromInt(8081)}}
+				svc.Spec.Selector = labels
+				svc.SetAnnotations(map[string]string{"loadbalancer.openstack.org/load-balancer-id": loadBalancerId,
+					"service.beta.kubernetes.io/openstack-internal-load-balancer": "true"})
+			})
+			o.Expect(err).To(o.HaveOccurred()) //Error is expected
+			o.Expect(svc2).To(o.BeNil())
+			g.By("Waiting for event reporting the error creating the internal lb")
+			eventSelector := fields.Set{
+				"involvedObject.kind":      "Service",
+				"involvedObject.name":      svcName2,
+				"involvedObject.namespace": oc.Namespace(),
+				"reason":                   "SyncLoadBalancerFailed",
+			}.AsSelector().String()
+			msg := "Error syncing load balancer: failed to ensure load balancer: internal Service cannot share a load balancer"
```

Comment:
Oh wow, I don't think we should depend on the message too much, any change there (like adapting to a new logging format) would invalidate the test for us. Moreover it'll be increasingly hard to keep it up to date if message changes in some CPO version.

Can we at a minimum only look for the "internal Service cannot share a load balancer" part?

Reply:
Right! I have updated to look for "internal Service cannot share a load balancer" . Thanks!